Source/WebCore/Modules/reporting/ReportingObserver.h

```diff
@@ -57,13 +57,19 @@ class ReportingObserver final : public RefCounted<ReportingObserver>, public Con
 
     ReportingObserverCallback& callbackConcurrently();
 
+    // ActiveDOMObject
+    void ref() const final { RefCounted::ref(); }
+    void deref() const final { RefCounted::deref(); }
+    bool virtualHasPendingActivity() const final { return m_observing; }
```

Comment:
Rather than holding a local `m_observing` bool, it might be more explicitly correct to ask `m_reportingScope` if the current object is registered, but the cost of that lookup is probably high and the chance of this boolean becoming stale/incorrect is probably pretty low.

Reply:
I could do a `Vector::find` from `ReportingScope`. That would save us the ugly bool.

Reply:
I could do a `Vector::find` from `ReportingScope`. That would save us the ugly bool.

Reply:
I did a "stress test" where I added 10k other ReportingObservers to the ReportingScope and there wasn't a significant impact to the test execution. So looking through a vector is probably OK and more maintainable than a bool.